velox/expression/fuzzer/ExpressionFuzzerVerifier.cpp

```diff
@@ -395,26 +399,41 @@ void ExpressionFuzzerVerifier::go() {
             {&execCtx_, {false, ""}, referenceQueryRunner_},
             *vectorFuzzer_,
             plans,
-            rowVector,
-            std::nullopt,
+            inputTestCases,
             inputRowMetadata);
       }
       throw;
     }
 
-    if (result.exceptionPtr) {
+    // If both paths threw compatible exceptions, we add a try() function to
+    // the expression's root and execute it again. This time the expressions
+    // cannot throw. Expressions that throw UNSUPPORTED_INPUT_UNCATCHABLE
+    // errors are not supported.
+    std::vector<fuzzer::InputTestCase> inputsToRetry;
+    bool anyInputsThrew = false;
+    bool anyInputsThrewButRetryable = false;
+    for (int j = 0; j < results.size(); j++) {
+      auto& result = results[j];
+      if (result.exceptionPtr) {
+        anyInputsThrew = true;
+        if (!result.unsupportedInputUncatchableError && options_.retryWithTry) {
+          anyInputsThrewButRetryable = true;
+          inputsToRetry.push_back(inputTestCases[j]);
+        }
+      } else {
+        // If we re-try then also run these inputs to ensure the conditions
+        // during test run stay close to original, that is, multiple inputs are
+        // executed.
+        inputsToRetry.push_back(inputTestCases[j]);
+      }
```

Comment:
I meant to keep it as an else branch for 417. The idea behind this was that if the eval did not encounter an exception, then it would always be added to the list of re-try, so that it can run alongside inputs that ran into an exception.

For eg, 2 input batches were executed, the first one ran fine, but the next error'ed out, now I want to make sure that when we re-try, we run both, so that if the first iteration had any role to play in the exception being thrown then it could be repro'ed.

Reply:
Ah, got it. Thank you for the explanation!